api/cloud-resources/v1beta1/azureredisinstance_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2023.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	featuretypes "github.com/kyma-project/cloud-manager/pkg/feature/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RedisInstanceAzureConfigs struct {
+	// +optional
+	MaxClients string `json:"maxclients,omitempty"`
+	// +optional
+	MaxFragmentationMemoryReserved string `json:"maxfragmentationmemory-reserved,omitempty"`
+	// +optional
+	MaxMemoryDelta string `json:"maxmemory-delta,omitempty"`
+	// +optional
+	MaxMemoryPolicy string `json:"maxmemory-policy,omitempty"`
+	// +optional
+	MaxMemoryReserved string `json:"maxmemory-reserved,omitempty"`
+	// +optional
+	NotifyKeyspaceEvents string `json:"notify-keyspace-events,omitempty"`
+	// +optional
+	ZonalConfiguration string `json:"zonal-configuration,omitempty"`
+}
+
+type RedisAuthSecretSpec struct {
+	Name        string            `json:"name,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+type AzureRedisSKU struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=1;2;3;4
+	Capacity int `json:"capacity"`
+}
+
+// AzureRedisInstanceSpec defines the desired state of AzureRedisInstance
+type AzureRedisInstanceSpec struct {
+	// +kubebuilder:validation:Required
+	SKU AzureRedisSKU `json:"sku"`
+
+	// +optional
+	EnableNonSslPort bool `json:"enableNonSslPort,omitempty"`
+
+	// +optional
+	RedisConfiguration RedisInstanceAzureConfigs `json:"redisConfiguration"`
+
+	// +optional
+	RedisVersion string `json:"redisVersion,omitempty"`
+
+	// +optional
+	ShardCount int `json:"shardCount,omitempty"`
+
+	// +optional
+	ReplicasPerPrimary int `json:"replicasPerPrimary,omitempty"`
+
+	AuthSecret *RedisAuthSecretSpec `json:"volume,omitempty"`
```

Comment:
If you are not handling mutation of these field, they should be marked immutable.
Also, consider adding default values to optional fields

Reply:
Fields set to immutable for now. Also, some of the default values are added.